libbeat/common/field.go

```diff
@@ -0,0 +1,126 @@
+package common
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/elastic/go-ucfg/yaml"
+)
+
+//This reflects allowed attributes for field definitions in the fields.yml.
+//No logic is put into this data structure.
+//The purpose is to enable using different kinds of transformation, on top of the same data structure.
+//Current transformation:
+//  -Kibana Index Pattern
+//  -ElasticSearch Template
+
+type Fields []Field
+
+type Field struct {
```

Comment:
actually long-term I think it would be nice to namespace kibana specific fields, but right now this would break even more.. 

Will add a comment for now.